extensions/hdrhistogram/src/main/java/org/radargun/stats/HistogramOperationStats.java

```diff
@@ -250,4 +249,62 @@ private void readObject(ObjectInputStream s) throws IOException, ClassNotFoundEx
       }
       compacted = new Histogram(ranges, counts);
    }
+
+   public final class InternalHistogram {
+
+      private long maxValue;
+      private int digits;
+
+      private org.HdrHistogram.Histogram responseTime;
+      private org.HdrHistogram.Histogram serviceTime;
+      private org.HdrHistogram.Histogram waitTime;
+
+      public InternalHistogram(long maxValue, int digits) {
+         this.maxValue = maxValue;
+         this.digits = digits;
+
+         responseTime = new org.HdrHistogram.Histogram(maxValue, digits);
+         serviceTime = new org.HdrHistogram.Histogram(maxValue, digits);
+         waitTime = new org.HdrHistogram.Histogram(maxValue, digits);
+      }
+
+      public void recordValue(long startTimeNs, long endTimeNs, long intendedTimeNs) {
+         serviceTime.recordValue(endTimeNs - startTimeNs);
+
+         if (intendedTimeNs != 0) { // if intended time is set we can compute response/wait
+            responseTime.recordValue(endTimeNs - intendedTimeNs);
+            waitTime.recordValue(startTimeNs - intendedTimeNs);
+         }
+      }
+
+      private org.HdrHistogram.Histogram getLatencyHistogram() {
+         if (responseTime.getTotalCount() == 0)
```

Comment:
If you always present just one of the histograms there's no reason to occupy memory by collecting the other histogram.
It seems to me that you could completely drop `intendedTime` and fill `Request.requestStart` with the value you pass to `indendedTime`, if the test is configured for such compensation, and that would save you the hassle with rewriting statistics.

Reply:
I don't believe that we can drop `intendedTime`.
The missing part in RadarGun according with https://psy-lob-saw.blogspot.com.br/2016/07/fixing-co-in-cstress.html is the notion of schedule.
For example:
intendedTime = 10:30
requestStartTime = 10:32
responseCompleteTime = 10:33
Per my understanding, we will have 3 minutes stored in InternalHistogram.responseTime instead of 1 minute.

Well, this was what I understood.

Reply:
If you never *use* requestStartTime, you don't need to record it at all, and I suggest using this field to store `intededTIme`.

Reply:
At this time, I would like to have both because I would like to have a chance to compare the values if we found something weird.
`intendedTime` for `responseTime`
`requestStartTime` for `serviceTime`

If this is not needed, I can remove `serviceTime` and `requestStartTime`